com.unity.multiplayer.mlapi/Runtime/NetworkVariable/NetworkVariable.cs

```diff
@@ -145,14 +145,6 @@ public bool CanClientRead(ulong clientId)
         /// <param name="stream">The stream to write the value to</param>
         public void WriteDelta(Stream stream)
```

Comment:
similar to what we are doing here, can we have `ReadDelta(Stream stream, bool keepDirtyDelta)` instead of `ReadDelta(Stream stream, bool keepDirtyDelta, ushort localTick, ushort remoteTick)` and push these down `localTick`/`remoteTick` reading from `NetworkBehaviour.HandleNetworkVariableDeltas(...)` method body.

Reply:
That is an awesome point and forced be to go back to re-thinking the whole rationale.

Where we start from:
- a set of individual Netvars that individually travel between machines.

Where we want to go:
- a snapshot system where a single message carries all the vars for a given frame. This message would have a single local tick (when it is sent) and many remote tick (when each individual var was modified). The dictionaries, sets and lists would (could?) go in too.

This experimental release, sadly, sits somewhere in between. Netvars (all types) looks like they carry remote tick and local tick. But only NetworkedVariables actually do. But we don't have the full snapshot system either.

I'm unwilling to revert back the whole tick transport into the NetworkedVariables just to not change stuff. In the end, the NetworkBehaviour will be an integral part of writing the snapshot. So, it feels OK to have some of the feature there. Similarly, in the end, the sets, lists, and dictionaries will have to be able to fit in the snapshot, so... This change is needed.
 
It's not optimal and, in hindsight, the part that goes in the experimental release would have been cut out differently. 

But for this release, I'd go forward with this PR, adding the tests that Matt suggests below. Would that be acceptable ?